app/models/cart.rb

```diff
@@ -0,0 +1,31 @@
+class Cart < ApplicationRecord
+  include Validations
+  belongs_to :coupon, optional: true
+  has_many :line_items, dependent: :destroy
+
+  def add_product(product, quantity)
+    return unless positive_integer?(quantity)
+    current_item = line_items.find_by(book_id: product.id)
+    if current_item
+      current_item.quantity += Integer(quantity)
+    else
+      current_item = line_items.build(book_id: product.id, quantity: quantity)
+    end
+    current_item
+  end
+
+  def subtotal
+    return 0 unless line_items.any?
+    line_items.sum(&:subtotal)
```

Comment:
@b-artem Prefer to use sql query over enumerable methods

Reply:
@Dulenchik, `subtotal` isn't present in DB, it's instance method only. Is it better to use SQL anyway or it should be presented in DB?